src/nanoemoji/maximum_color.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generates color tables to provide support for a wide range of browsers.
+
+Requires input font have vector color capability, that is either an SVG table
+or COLR/CPAL.
+
+https://www.youtube.com/watch?v=HLddvNiXym4
+
+Sample usage:
+
+python -m nanoemoji.maximum_color MySvgFont.ttf"""
+from absl import app
+from absl import flags
+from absl import logging
+from fontTools import ttLib
+from nanoemoji.extract_svgs import svg_glyphs
+from nanoemoji.ninja import (
+    build_dir,
+    gen_ninja,
+    maybe_run_ninja,
+    module_rule,
+    rel_build,
+    NinjaWriter,
+)
+from nanoemoji.util import only
+from pathlib import Path
+
+
+_SVG2COLR_GLYPHMAP = "svg2colr.glyphmap"
+_SVG2COLR_CONFIG = "svg2colr.toml"
+
+
+FLAGS = flags.FLAGS
+
+
+flags.DEFINE_bool(
+    "destroy_non_color_glyphs",
+    True,
+    "If true feel free to obliterate any existing glyf/cff content, e.g. fallback glyphs",
+)
+
+
+def _vector_color_table(font: ttLib.TTFont) -> str:
+    has_svg = "SVG " in font
+    has_colr = "COLR" in font
+    if has_svg == has_colr:
+        raise ValueError("Must have one of COLR, SVG")
+
+    if has_svg:
+        return "SVG "
+    if has_colr:
+        return "COLR"
+
+    raise ValueError("Impossible")
+
+
+def svg_extract_dir() -> Path:
+    return build_dir() / "svg_dump"
+
+
+def picosvg_dir() -> Path:
+    return build_dir() / "picosvg"
+
+
+def picosvg_dest(input_svg: Path) -> Path:
+    return picosvg_dir() / input_svg.name
+
+
+def _write_preamble(nw: NinjaWriter, input_font: Path):
+    module_rule(
+        nw,
+        "extract_svgs_from_otsvg",
+        f"--output_dir {rel_build(svg_extract_dir())} $in $out",
+    )
+    nw.newline()
+
+    module_rule(nw, "write_glyphmap_for_glyph_svgs", "--output_file $out $in")
```

Comment:
in the write_glyphmap_for_glyph_svgs.py you are calling util.expand_ninja_response_file method (rightly so because picosvgs inputs can be in the thousands and likely to break shell args limits) however here in the rule definition you forgot to use the two `rspfile` and `rspfile_content` parameters (see `write_glyphmap_rule` for an example in nanoemoji/nanoemoji.py).
You need to do that otherwise there's no .rsp file to expand, and things will break with fonts with many glyphs like noto-emoji

Reply:
Ew. Good catch, ty.